src/java/org/apache/cassandra/io/sstable/format/big/BigTableWriter.java

```diff
@@ -80,14 +80,23 @@ public class BigTableWriter extends SortedTableWriter<BigFormatPartitionWriter>
     public BigTableWriter(Builder builder, LifecycleNewTracker lifecycleNewTracker, SSTable.Owner owner)
     {
         super(builder, lifecycleNewTracker, owner);
-        checkNotNull(builder.getRowIndexEntrySerializer());
-        checkNotNull(builder.getIndexWriter());
-
-        this.rowIndexEntrySerializer = builder.getRowIndexEntrySerializer();
-        this.indexWriter = builder.getIndexWriter();
-        this.shouldMigrateKeyCache = DatabaseDescriptor.shouldMigrateKeycacheOnCompaction()
-                                     && lifecycleNewTracker instanceof ILifecycleTransaction
-                                     && !((ILifecycleTransaction) lifecycleNewTracker).isOffline();
+
+        try
+        {
+            this.rowIndexEntrySerializer = builder.getRowIndexEntrySerializer();
+            this.indexWriter = builder.getIndexWriter().get();
+
+            checkNotNull(builder.getRowIndexEntrySerializer());
+            checkNotNull(builder.getIndexWriter());
```

Comment:
I don't think this is doing the intended check, `this.indexWriter` is the one that should not be null.

Reply:
right, my bad, moved checkNotNull but forgot to modify the arg